filter.go

```diff
@@ -242,8 +244,31 @@ func compileFilter(filter string, pos int) (*ber.Packet, int, error) {
 			}
 			packet.AppendChild(seq)
 		default:
+			var buffer bytes.Buffer
+			for i := 0; i < len(condition); i++ {
+				// Check for escaped hex characters and convert them to their literal value for transport.
+				if condition[i] == '\\' {
+					// http://tools.ietf.org/search/rfc4515
+					// \ (%x5C) is not a valid character unless it is followed by two HEX characters due to not
+					// being a member of UTF1SUBSET.
+					if i+2 > len(condition) {
+						err = NewError(ErrorFilterCompile, errors.New("ldap: missing characters for escape in filter"))
+						return packet, newPos, err
+					}
+					if escByte, decodeErr := hexpac.DecodeString(condition[i+1:i+3]); decodeErr != nil {
```

Comment:
isn't this subject to panic if i+1 or i+3 is greater than the length of the condition?